src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,46 +496,67 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
      * </pre>
      *
      * @param currentCase AST of the case that falls through to the next case.
-     * @param nextCase AST of the next case.
      * @return True if a relief comment was found
      */
-    private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase) {
-        boolean allThroughComment = false;
-        final int endLineNo = nextCase.getLineNo();
-
-        if (matchesComment(reliefPattern, endLineNo)) {
-            allThroughComment = true;
+    private boolean hasFallThroughComment(DetailAST currentCase) {
+        final DetailAST nextCase = currentCase.getNextSibling();
+        final DetailAST ast;
+        final DetailAST lastAST;
+        if (nextCase.getType() == TokenTypes.CASE_GROUP) {
+            ast = nextCase;
+            lastAST = nextCase.getLastChild();
         }
         else {
-            final int startLineNo = currentCase.getLineNo();
-            for (int i = endLineNo - 2; i > startLineNo - 1; i--) {
-                final int[] line = getLineCodePoints(i);
-                if (!CodePointUtil.isBlank(line)) {
-                    allThroughComment = matchesComment(reliefPattern, i + 1);
-                    break;
-                }
-            }
+            ast = currentCase;
+            lastAST = currentCase.getParent().getLastChild();
         }
-        return allThroughComment;
+        final List<DetailAST> commentContent = getCommentContentNodes(ast, lastAST);
+        final List<DetailAST> commentAsChildOfSameGroup =
+                getCommentContentNodes(currentCase.getLastChild(),
+                        currentCase.getLastChild().getLastChild());
```

Comment:
I still does not understand why we need to do search twice ?
if you have 4 Ast, can you make methods that take "overlap" of them and return range (from/to) that we need.
To make it clear we need all comments in this range of AST nodes.

Reply:
I have removed this totally we have to check from the current case's last to next case last child

done